maxim/logger/llamaindex/__init__.py

```
@@ -0,0 +1,114 @@
+    def end_trace(self, trace_id: Optional[str] = None, trace_map: Optional[Dict[str, List[str]]] = None) -> None:
+        pass
+
+    def _parse_messages(self, messages: Sequence[ChatMessage]) -> List[Dict[str, str]]:
+        parsed: List[Dict[str, str]] = []
+        for msg in messages:
+            role = msg.role.value if hasattr(msg.role, "value") else str(msg.role)
+            content = str(msg.content) if msg.content is not None else ""
+            parsed.append({"role": role, "content": content})
+        return parsed
+
+    def _parse_provider(self, class_name: Optional[str]) -> str:
```

Ruff (0.11.9)

Use `X | Y` for type annotations

Convert to `X | Y`

(UP007)